cmd/proxygenerator/go.mod

```
@@ -1,6 +1,8 @@
 module go.temporal.io/api/cmd/proxygenerator
 
-go 1.20
+go 1.21
+
+toolchain go1.22.3
```

While I agree with @cretz, you can replace this with just:
```suggestion
go 1.22.3
```

---

No, that would be even worse. That would tell people they need at least 1.22.3 to use this. This libraries is the most depended upon user library in Temporal and we should at least support all non-EOL Go versions (but ideally we don't change any of this unless there is a good reason, and even if there is a good reason, not sure it should be in a makefile-related PR).